ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/DerivationPath.scala

```diff
@@ -57,6 +61,32 @@ object DerivationPath {
     }
   }
 
+  /**
+    * Finds next available path index for a new key.
+    */
+  def nextPath(secrets: IndexedSeq[ExtendedSecretKey]): Try[DerivationPath] = {
```

Comment:
Please describe what is going on in this method, the logic is not comprehensible.

Reply:
This method is not part of this PR, it was just moved from other place. Honestly, some efforts would be needed to understand it, along with making tests. Made logic description part of this PR: https://github.com/ergoplatform/ergo/issues/1082 .